backend/dataall/modules/dataset_sharing/services/share_notification_service.py

```diff
@@ -96,3 +128,33 @@ def _get_dataset_stewards(dataset):
         stewards.append(dataset.SamlAdminGroupName)
         stewards.append(dataset.stewards)
         return stewards
+
+    @staticmethod
+    def create_notification_task(session, email_id, dataset, share, subject, msg):
+        share_notification_config = config.get_property('core.share_notifications')
+        requester_group_name = share.groupUri
+        dataset_owner_group = dataset.SamlAdminGroupName
+        dataset_stewards_group = dataset.stewards
+
+        for share_notification_config_type in share_notification_config.keys():
+            if share_notification_config_type == 'email' and share_notification_config[share_notification_config_type]['active'] == True:
+
+                notification_task: Task = Task(
+                    action='notification.service',
+                    targetUri=share.shareUri,
+                    payload={
+                        'notificationType' : share_notification_config_type,
+                        'shareRequestUserEmail': email_id,
+                        'subject': subject,
+                        'message': msg,
+                        'requesterGroupName': requester_group_name,
```

Comment:
Ideally the payload does not include any logic relative to the datasets and datasharing. Since we already have created the message `requesterGroupName`, `datasetOwnerGroup` and `datasetStewardsGroup` could be passed as a single field `recipients`. We will need to do the check `notification_configs['email']['features']['group_notifications'] == True:` in email_notification_provider line 56 in the share_notification service

Reply:
As discussed made changes in the next commit